modules/imgproc/src/smooth.cpp

```diff
@@ -2304,22 +2319,25 @@ void hlineSmooth5N<uint8_t, ufixedpoint16>(const uint8_t* src, int cn, const ufi
 
         src += 2 * cn; dst += 2 * cn;
         int i = 2*cn, lencn = (len - 2)*cn;
-        v_uint16x8 v_mul0 = v_setall_u16(*((uint16_t*)m));
-        v_uint16x8 v_mul1 = v_setall_u16(*((uint16_t*)(m + 1)));
-        v_uint16x8 v_mul2 = v_setall_u16(*((uint16_t*)(m + 2)));
-        v_uint16x8 v_mul3 = v_setall_u16(*((uint16_t*)(m + 3)));
-        v_uint16x8 v_mul4 = v_setall_u16(*((uint16_t*)(m + 4)));
-        for (; i <= lencn - 16; i += 16, src += 16, dst += 16)
-        {
-            v_uint16x8 v_src00, v_src01, v_src10, v_src11, v_src20, v_src21, v_src30, v_src31, v_src40, v_src41;
-            v_expand(v_load(src - 2*cn), v_src00, v_src01);
-            v_expand(v_load(src - cn), v_src10, v_src11);
-            v_expand(v_load(src), v_src20, v_src21);
-            v_expand(v_load(src + cn), v_src30, v_src31);
-            v_expand(v_load(src + 2*cn), v_src40, v_src41);
+#if CV_SIMD
+        const int VECSZ = v_uint16::nlanes;
+        v_uint16 v_mul0 = vx_setall_u16(*((uint16_t*)m));
+        v_uint16 v_mul1 = vx_setall_u16(*((uint16_t*)(m + 1)));
+        v_uint16 v_mul2 = vx_setall_u16(*((uint16_t*)(m + 2)));
+        v_uint16 v_mul3 = vx_setall_u16(*((uint16_t*)(m + 3)));
+        v_uint16 v_mul4 = vx_setall_u16(*((uint16_t*)(m + 4)));
```

Comment:
May be add `.as_uint16()` (or something like `.reinterpter_as_u16()`) method?
This will be look more readable:
```
m[0].as_uint16()
m[1].as_uint16()
m[2].as_uint16()
m[3].as_uint16()
m[4].as_uint16()
```

Reply:
@alalek, not needed; I would rather cast `m` pointer to `uint16_t*` in the beginning